File: 0066-plus-one/0066-plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        for i in range(len(digits)):
            if digits[-i - 1] == 9:
                digits[-i - 1] = 0
            else:
                digits[-i - 1] += 1
                break
                
        if digits[0] == 0:
            return [1] + digits
        else:
            return digits

File: 0066-plus-one/0066-plus-one.kt
class Solution {
    fun plusOne(digits: IntArray): IntArray {
        for (i in digits.size - 1 downTo 0) {
            if (digits[i] == 9) {
                digits[i] = 0
            } else {
                digits[i] += 1
                return digits
            }
        }
        
        return intArrayOf(1) + digits
    }
}